packages/grid/x-data-grid/src/components/cell/GridCell.tsx

```diff
@@ -125,16 +107,111 @@ const GridCell = React.forwardRef<HTMLDivElement, GridCellProps>((props, ref) =>
     ...other
   } = props;
 
-  const valueToRender = formattedValue == null ? value : formattedValue;
+  const field = column.field;
   const cellRef = React.useRef<HTMLDivElement>(null);
   const handleRef = useForkRef(ref, cellRef);
   const focusElementRef = React.useRef<FocusElement>(null);
   const apiRef = useGridApiContext();
-
   const rootProps = useGridRootProps();
+
+  const editRowsState = useGridSelector(apiRef, gridEditRowsStateSelector);
+
+  const { classes: rootClasses, getCellClassName } = rootProps;
+
+  const isSelected = useGridSelector(apiRef, () =>
+    apiRef.current.unstable_applyPipeProcessors('isCellSelected', false, {
+      id: rowId,
+      field,
+    }),
+  );
```

Comment:
Some of our `state` accesses are hidden behind pipe processors, this is why some selectors don't use their input parameter.

Reply:
At first, I didn't understand why `useGridSelector` is used here.
But now I see 2 reasons:
1. To avoid calling `unstable_applyPipeProcessors` on each rerender
2. To call `unstable_applyPipeProcessors` on state change even if the cell is not rerendered (because of React.memo)

Do I understand it correctly?

Reply:
It's used to create a reactive binding so that when the state changes, `isSelected` is recomputed, which trigger a re-render of the cell if it changed. Underneath, the pipe processor is [accessing the state here](https://github.com/mui/mui-x/blob/138dba77936385405ec92688aaabe323c74deb05/packages/grid/x-data-grid-premium/src/hooks/features/cellSelection/useGridCellSelection.ts#L83-L89).